### PR TITLE
Save subscription flow boilerplate

### PR DESCRIPTION
**Pull request by sohanirao**

### CODE REVIEW

File: src/runtime/save-subscription-flow.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    /** @private @const {!ActivityIframeView} */
+    this.activityIframeView_ = new ActivityIframeView(
+        this.win_,
+        this.activityPorts_,
+        feUrl('/linksaveiframe'),
+        feArgs({
+          'publicationId': deps.pageConfig().getPublicationId(),
+        }),
+        /* shouldFadeBody */ true
```
Comment (dvoytenko) [inline src/runtime/save-subscription-flow.js]: I don't think it should fade - an optional feature overall.

Comment (sohanirao) [reply src/runtime/save-subscription-flow.js]: Done

File: src/runtime/link-accounts-flow.js [hunk 2]
```diff
@@ -242,3 +242,71 @@ export class LinkCompleteFlow {
...
+    /** @private @const {!Window} */
+    this.win_ = deps.win();
+
+    /** @private @const {!web-activities/activity-ports.ActivityPorts} */
+    this.activityPorts_ = deps.activities();
+
+    /** @private @const {!../components/dialog-manager.DialogManager} */
+    this.dialogManager_ = deps.dialogManager();
+
+    /** @private {?string} */
```
[the remark this answers is not in the document]
Comment (sohanirao) [reply src/runtime/link-accounts-flow.js]: Done

File: src/runtime/link-accounts-flow.js [hunk 3]
```diff
@@ -242,3 +242,71 @@ export class LinkCompleteFlow {
...
 }
+
+/**
+ * The flow to save subscription information.
+ */
+export class LinkSaveFlow {
+
+  /**
+   * @param {!./deps.DepsDef} deps
+   * @param {!(../api/subscriptions.Token|function():!string)} tokenOrCallback
```
[the remark this answers is not in the document]
Comment (sohanirao) [reply src/runtime/link-accounts-flow.js]: Not sure I understand. subscriptions.js has the new typedef SaveSubscriptionRequest.

File: src/runtime/link-accounts-flow-test.js [hunk 4]
```diff
@@ -259,3 +260,45 @@ describes.realWin('LinkCompleteFlow', {}, env => {
...
+  let linkSaveFlow;
+  let port;
+
+  beforeEach(() => {
+    win = env.win;
+    pageConfig = new PageConfig('pub1:label1');
+    runtime = new ConfiguredRuntime(win, pageConfig);
+    activitiesMock = sandbox.mock(runtime.activities());
+    callbacksMock = sandbox.mock(runtime.callbacks());
+    linkSaveFlow = new LinkSaveFlow(runtime, {'token': 'test'});
```
[the remark this answers is not in the document]
Comment (sohanirao) [reply src/runtime/link-accounts-flow-test.js]: done

File: src/api/subscriptions.js [hunk 5]
```diff
@@ -138,3 +144,11 @@ export let OptionsRequest;
  * }}
  */
 export let LoginRequest;
+
```
[the remark this answers is not in the document]
Comment (sohanirao) [reply src/api/subscriptions.js]: done

File: src/api/subscriptions.js [hunk 6]
```diff
@@ -138,3 +144,11 @@ export let OptionsRequest;
...
+/**
+ * @typedef {{
+ *   token: !string,
```
[the remark this answers is not in the document]
Comment (sohanirao) [reply src/api/subscriptions.js]: done

File: src/api/subscriptions.js [hunk 7]
```diff
@@ -114,6 +114,12 @@ export class Subscriptions {
    * TODO(dparikh): decide if it's only exposed for testing or PROD purposes.
    */
   linkAccount() {}
+
+  /**
+   * Starts the save subscriptions flow.
+   * @param {!(Token|function():!string)} tokenOrCallback
```
[the remark this answers is not in the document]
Comment (sohanirao) [reply src/api/subscriptions.js]: done

File: docs/link-flow.md [hunk 8]
```diff
@@ -19,3 +19,15 @@ limitations under the License.
 This flow is normally originated from another surface and allows the reader to link this publication's subscription to that surface. See [Subscriptions APIs](./core-apis.md).
 
 The link flow would normally be executed by a 3p surface to let a user to claim an existing subscription. However, SwG client provides `linkAccount` and `setOnLinkComplete` APIs for testing purposes.
+
+# SwG Link Save flow
```
[the remark this answers is not in the document]
Comment (sohanirao) [reply docs/link-flow.md]: done

[the remark this answers is not in the document]
Comment (dvoytenko) [reply docs/link-flow.md]: Not seeing this. Did you push?

File: src/api/subscriptions.js [hunk 9]
```diff
@@ -114,6 +114,12 @@ export class Subscriptions {
    * TODO(dparikh): decide if it's only exposed for testing or PROD purposes.
    */
   linkAccount() {}
+
+  /**
+   * Starts the save subscriptions flow.
+   * @param {!SaveSubscriptionRequest} saveSubscriptionRequest
```
Comment (dvoytenko) [inline src/api/subscriptions.js]: Change arg name to just `request`

File: src/api/subscriptions.js [hunk 10]
```diff
@@ -114,6 +114,12 @@ export class Subscriptions {
    * TODO(dparikh): decide if it's only exposed for testing or PROD purposes.
    */
   linkAccount() {}
+
+  /**
+   * Starts the save subscriptions flow.
+   * @param {!(Token|function():!string)} tokenOrCallback
+   */
```
[the remark this answers is not in the document]
Comment (dvoytenko) [reply src/api/subscriptions.js]: This still needs to be done.

File: src/runtime/link-accounts-flow.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (dvoytenko) [reply src/runtime/link-accounts-flow.js]: This is good now. I just meant to use the `SaveSubscriptionRequest `.

File: docs/link-flow.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (sohanirao) [reply docs/link-flow.md]: yes this one's uploaded. can you open files and see the latest one?

File: src/api/subscriptions.js (hunk 9 above)
Comment (sohanirao) [reply src/api/subscriptions.js]: done

File: docs/link-flow.md [hunk 11]
```diff
@@ -19,3 +19,15 @@ limitations under the License.
...
 
 The link flow would normally be executed by a 3p surface to let a user to claim an existing subscription. However, SwG client provides `linkAccount` and `setOnLinkComplete` APIs for testing purposes.
+
+# SwG Link Save flow (Not launched)
+
+This flow is normally originated from the publisher and allows the reader to link publication's subscription to the reader's account.
+
+To start saving subscription link, provide a request token corresponding to the readers subscription:
+
+subscriptions.saveSubscription(saveSubscriptionRequest);
```
Comment (dvoytenko) [inline docs/link-flow.md]: Add code section for this. Also please expand what request looks like. E.g. `{token: 'THE TOKEN'}`.

File: docs/link-flow.md [hunk 12]
```diff
@@ -19,3 +19,18 @@ limitations under the License.
 This flow is normally originated from another surface and allows the reader to link this publication's subscription to that surface. See [Subscriptions APIs](./core-apis.md).
 
 The link flow would normally be executed by a 3p surface to let a user to claim an existing subscription. However, SwG client provides `linkAccount` and `setOnLinkComplete` APIs for testing purposes.
+
+# SwG Link Save flow (Not launched)
+
+This flow is normally originated from the publisher and allows the reader to link publication's subscription to the reader's account.
+
+To start saving subscription link, provide a request containing a token corresponding to the reader's
+subscription, Eg: {token: 'THE TOKEN'}
```
Comment (dvoytenko) [inline docs/link-flow.md]: Not quite. I meant actually use `{token: 'THE TOKEN'}` in the example below inside the code section. E.g.

```
subscriptions.saveSubscription({token: 'THE TOKEN'});
```

**Pull request closed without merging**